### ISSUE TITLE
Dynamically alter behaviour when on single node cluster

### ISSUE THREAD

**Issue by benrutter** [dask-contrib/dask-databricks#39]
I noticed that when on a databricks single node cluster, that after running:

```python
client = dask_databricks.get_client()
```

Dask calls will just hang - this makes sense since there is only a driver and no worker nodes, so nothing to actually do the work.

It would be really nice if:
- After a while of waiting, the cluster threw an error because no workers are available (I'm guessing this would belong in `dask/distributed` proper though?)
- dask-databricks had some kind of clever behaviour to use multi-process rather than distributed workflows in on a single cluster node

I think the obvious way to implement the second would be to have some behaviour change based on some kind of `spark.conf` variable (although I haven't seen an obvious candidate).

@jacobtomlinson, any thoughts? Would this be a handy feature or is it a bit of an overcomplication? (I can't figure thinking it through if its actually worth the additional complexity - it'd probably be handy if someone had a notebook that they wanted to run over multiple different cluster types, but the use case for `dask-databricks` isn't really there?)

**Comment by jacobtomlinson** [comment]
I think it would be good to try and detect a single node cluster in `get_client()`. However I think I would err on the side of throwing an exception rather than doing something magic under the hood. Then you could catch it and intentionally choose what to do. 

Perhaps something like this:

```python
import dask_databricks
from dask.distributed import LocalCluster

try:
    client = dask_databricks.get_client()
except dask_databricks.SingleNodeException:
    client = LocalCluster().get_client()
```

**Comment by benrutter** [comment]
I like the idea of throwing an exception a lot! I think that solves the main issue which is that if you're not familiar with dask you might get confused/frustrated with it just hanging, without realising why that's happening.

I'll have a look at what information is available from spark.conf and hopefully put in a PR

**Comment by jacobtomlinson** [comment]
Sounds great. I look forward to reviewing it. There may also be uneful information just stored in environment variables too.

**Comment by jacobtomlinson** [comment]
This is now available in `0.3.2`
